lib/byron/test/integration/Cardano/Wallet/Byron/Config.hs

```diff
@@ -0,0 +1,251 @@
+    :: (  CardanoNodeConfig
+       -> Block
+       -> (BlockchainParameters, NodeVersionData)
+       -> IO a
+       )
+    -- ^ Callback function with the node configuration and genesis params
+    -> IO a
+withConfig action =
+    bracket setupConfig teardownConfig $ \(_a,b,c,d) -> action b c d
+  where
+    source :: FilePath
+    source = $(getTestData) </> "cardano-node"
+
+    setupConfig
+        :: IO ( FilePath
+              , CardanoNodeConfig
+              , Block
+              , (BlockchainParameters, NodeVersionData)
+              )
+    setupConfig = do
+        dir <- getCanonicalTemporaryDirectory
+            >>= \tmpRoot -> createTempDirectory tmpRoot "cardano-wallet-byron"
+
+        let nodeConfigFile   = dir </> "node.config"
+        let nodeDatabaseDir  = dir </> "node.db"
+        let nodeDlgCertFile  = dir </> "node.cert"
+        let nodeGenesisFile  = dir </> "genesis.json"
+        let nodeSignKeyFile  = dir </> "node.key"
+        let nodeSocketFile   = dir </> "node.socket"
+        let nodeTopologyFile = dir </> "node.topology"
+
+        Yaml.decodeFileThrow @_ @Aeson.Value (source </> "genesis.yaml")
+            >>= withObject updateStartTime
+            >>= Aeson.encodeFile nodeGenesisFile
+        forM_ ["node.config", "node.topology", "node.key", "node.cert"] $ \f ->
+            copyFile (source </> f) (dir </> f)
+
+        (genesisData, genesisHash) <- unsafeRunExceptT $
+            readGenesisData nodeGenesisFile
+        let (bp, outs) = fromGenesisData (genesisData, genesisHash)
+
+        let networkMagic = NetworkMagic $ unProtocolMagicId $ gdProtocolMagicId genesisData
+        let nodeGenesisHash = T.decodeUtf8 $ hex $ getHash $ getGenesisBlockHash bp
+
+        pure
+            ( dir
+            , CardanoNodeConfig
+                { nodeGenesisHash
+                , nodeConfigFile
+                , nodeDatabaseDir
+                , nodeDlgCertFile
+                , nodeGenesisFile
+                , nodeSignKeyFile
+                , nodeSocketFile
+                , nodeTopologyFile
+                }
+            , mkBlock0 bp outs
+            , ( bp
+              , ( NodeToClientVersionData { networkMagic }
+                , nodeToClientCodecCBORTerm
+                )
+              )
+            )
+
+    teardownConfig
+        :: (FilePath, b, c, d)
+        -> IO ()
+    teardownConfig (dir, _, _, _) = do
+        noCleanup <- maybe False (not . null) <$> lookupEnv "NO_CLEANUP"
+        exists <- doesDirectoryExist dir
+        unless noCleanup $ when exists $ removeDirectoryRecursive dir
+
+-- | Add a "startTime" field in a given object with the current POSIX time as a
+-- value.
+updateStartTime
+    :: Aeson.Object
+    -> IO Aeson.Object
+updateStartTime m = do
+    time <- round @_ @Int <$> getPOSIXTime
+    pure $ HM.insert "startTime" (toJSON time) m
+
+-- | Do something with an a JSON object. Fails if the given JSON value isn't an
+-- object.
+withObject
```

Comment:
wonder if this should not go to Cardano.Wallet.Api.Types and then we would remove aeson dependency?

Reply:
Hmmm. Yeah, could be in a `Data.Aeson.Extra` possibly. The function is not unsafe per se if we instantiate the monad `m` to `Maybe` for example. Yet I am not sure it is quite needed anywhere else so we may as well move it if/when needed :thinking: ?

Reply:
ah, aeson is needed also for yaml processing..